packages/ast-spec/src/declaration/VariableDeclaration/spec.ts

```diff
@@ -36,18 +39,55 @@ export interface LetOrConstOrVarDeclaration extends BaseNode {
   kind: 'const' | 'let' | 'var';
 }
 
-export interface UsingInNormalContextDeclaration extends BaseNode {
-  type: AST_NODE_TYPES.VariableDeclaration;
+export interface LetOrVarDeclaredDeclaration
+  extends LetOrConstOrVarDeclarationBase {
+  kind: 'let' | 'var';
+  declare: true;
   /**
-   * The variables declared by this declaration.
-   * Note that there may be 0 declarations (i.e. `const;`).
+   * In a `declare let` declaration, the declarators must not have definite assignment
+   * assertions or initializers.
    * ```
-   * using x = 1;
-   * using y =1, z = 2;
+   * declare let x: number;
+   * declare let y, z;
    * ```
    */
-  // TODO(#1852) - this should be guaranteed to have at least 1 element in it.
-  declarations: UsingInNomalConextDeclarator[];
+  declarations: VariableDeclaratorNoInit[];
+}
+
+export interface LetOrVarNonDeclaredDeclaration
+  extends LetOrConstOrVarDeclarationBase {
+  kind: 'let' | 'var';
+  declare: false;
+  /**
+   * In a `let`/`var` declaration, the declarators may have definite assignment
+   * assertions or initializers, but not both.
+   */
+  declarations: (
+    | VariableDeclaratorDefiniteAssignment
+    | VariableDeclaratorMaybeInit
+  )[];
+}
+
+export interface ConstDeclaration extends LetOrConstOrVarDeclarationBase {
+  kind: 'const';
+  /**
+   * In a `declare const` declaration, the declarators may have initializers, but
+   * not definite assignment assertions. Each declarator cannot have both an
+   * initializer and a type annotation.
+   *
+   * Even if the declaration has no `declare`, it may still be ambient and have
+   * no initializer.
+   */
+  declarations: VariableDeclaratorMaybeInit[];
```

Comment:
```suggestion
  // TODO(#1852) - this should be guaranteed to have at least 1 element in it.
  declarations: VariableDeclaratorMaybeInit[];
```

Reply:
We already have this restriction in the converter, but as said above I don't think restricting the type here gives any value